src/libs/WorkspacesSettingsUtils.ts

```diff
@@ -119,14 +120,25 @@ function hasWorkspaceSettingsRBR(policy: Policy) {
     return Object.keys(reimbursementAccount?.errors ?? {}).length > 0 || hasPolicyError(policy) || hasCustomUnitsError(policy) || policyMemberError || taxRateError;
 }
 
-function getChatTabBrickRoadReport(policyID?: string): OnyxEntry<Report> {
-    const allReports = ReportConnection.getAllReports();
-    if (!allReports) {
+function getChatTabBrickRoadReport(
+    policyID: string | undefined,
+    currentReportId: string | null,
+    reports: OnyxCollection<Report>,
+    betas: OnyxEntry<Beta[]>,
+    policies: OnyxCollection<PolicySelector>,
+    priorityMode: OnyxEntry<PriorityMode>,
+    transactionViolations: OnyxCollection<TransactionViolation[]>,
+    policyMemberAccountIDs: number[] = [],
+): OnyxEntry<Report> {
+    const reportIDs = SidebarUtils.getOrderedReportIDs(currentReportId, reports, betas, policies, priorityMode, transactionViolations, policyID, policyMemberAccountIDs);
+    if (!reportIDs.length) {
         return undefined;
     }
 
+    const allReports = reportIDs.map((reportID) => reports?.[`${ONYXKEYS.COLLECTION.REPORT}${reportID}`]);
+
     // If policyID is undefined, then all reports are checked whether they contain any brick road
-    const policyReports = policyID ? Object.values(allReports).filter((report) => report?.policyID === policyID) : Object.values(allReports);
+    const policyReports = Object.values(allReports).filter((report) => !policyID || report?.policyID === policyID);
```

Comment:
@pac-guerreiro Why do you change this? I think the previous way would have been better. It avoids the loop if policyID is undefined and saves more condition checks. Wdyt?

Reply:
@DylanDylann You're totally right! I just looked at that line and thought I could simplify it 😅 

I'll just revert this change, thanks